src/main.jsx

```diff
@@ -2,9 +2,31 @@ import React from 'react'
 import ReactDOM from 'react-dom/client'
 import App from './App.jsx'
 import './index.css'
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import Root from "./routes/root.jsx";
+import ErrorPage from "./error-page.jsx";
+import { SubPage, RootSubpage } from "./routes/subpage.jsx";
+const router = createBrowserRouter([
```

Comment:
Nice! I personally like using [this](https://reactrouter.com/en/main/utils/create-routes-from-elements) to write the routes as JSX